Firestore/core/src/local/leveldb_persistence.cc

```diff
@@ -259,9 +275,33 @@ LevelDbBundleCache* LevelDbPersistence::bundle_cache() {
 LevelDbDocumentOverlayCache* LevelDbPersistence::GetDocumentOverlayCache(
     const User& user) {
   users_.insert(user.uid());
-  current_document_overlay_cache_ =
-      absl::make_unique<LevelDbDocumentOverlayCache>(user, this, &serializer_);
-  return current_document_overlay_cache_.get();
+  if (document_overlay_caches_.find(user.uid()) ==
+      document_overlay_caches_.end()) {
+    document_overlay_caches_.insert(
+        {user.uid(), absl::make_unique<LevelDbDocumentOverlayCache>(
+                         user, this, &serializer_)});
+  }
+  return document_overlay_caches_[user.uid()].get();
+}
+
+LevelDbOverlayMigrationManager* LevelDbPersistence::GetOverlayMigrationManager(
```

Comment:
If `GetOverlayMigrationManager()` is invoked with a different user than a previous invocation it's going to return the `LevelDbOverlayMigrationManager` instance associated with the _old_ user (the caller likely requires that the returned `LevelDbOverlayMigrationManager` instance is associated with the _new_ user).

All of the other methods in this class that take a `const credentials::User& user` argument also have a dictionary mapping user to the instance of the type being returned. Maybe this method should do that too?